### PR TITLE
Collapse Context and UpdateContext structs

### PR DESCRIPTION
**Pull request by EdNoepel**
For code size reduction.

### CODE REVIEW

**Comment by EdNoepel** [comment]
Ran all the integration tests locally; all pass.  Have rerun some of these CI jobs several times, but keep having `HttpProviderError` failures.  Recommend checking Alchemy dashboard to investigate.

File: packages/perennial/contracts/Market.sol [hunk 1]
```diff
@@ -409,42 +409,42 @@ contract Market is IMarket, Instance, ReentrancyGuard {
...
             collateral,
             newMaker,
             newLong,
             newShort,
             protect,
             referralFee
         );
-        updateContext.currentPosition.global.update(newOrder);
-        updateContext.currentPosition.local.update(newOrder);
+        updateContext.currentGlobalPosition.update(newOrder);
```
Comment (kbrizzle) [inline packages/perennial/contracts/Market.sol:440]: Could we keep the naming consistent for now: `currentPosition.local` -> `currentPositionLocal`.

File: packages/perennial/contracts/Market.sol [hunk 2]
```diff
@@ -409,42 +409,42 @@ contract Market is IMarket, Instance, ReentrancyGuard {
...
+        updateContext.currentLocalPosition.update(newOrder);
 
         // apply new order
-        updateContext.order.local.add(newOrder);
-        updateContext.order.global.add(newOrder);
-        context.pending.global.add(newOrder);
-        context.pending.local.add(newOrder);
+        updateContext.orderLocal.add(newOrder);
+        updateContext.orderGlobal.add(newOrder);
+        context.pendingGlobalOrder.add(newOrder);
```
Comment (kbrizzle) [inline packages/perennial/contracts/Market.sol:446]: Here as well `pending.global` -> `pendingGlobal`.

File: packages/perennial/contracts/Market.sol [hunk 3]
```diff
@@ -513,9 +513,9 @@ contract Market is IMarket, Instance, ReentrancyGuard {
         address account
     ) private view returns (SettlementContext memory settlementContext) {
         // processing accumulators
-        settlementContext.latestVersion = _versions[context.latestPosition.global.timestamp].read();
-        settlementContext.latestCheckpoint = _checkpoints[account][context.latestPosition.local.timestamp].read();
-        settlementContext.orderOracleVersion = oracle.at(context.latestPosition.global.timestamp);
+        settlementContext.latestVersion = _versions[context.latestGlobalPosition.timestamp].read();
```
Comment (kbrizzle) [inline packages/perennial/contracts/Market.sol:516]: And here `latestPosition.global` -> `latestPositionGlobal`.

**Comment by kbrizzle** [comment]
Left some nits on naming, I'm thinking we should keep things as is for now. We may want to readdress these after we complete the `currentPosition` cleanup task.

The intuition here is that there are three things we're keeping track of:
- `position` -> the settled position.
- `order` -> the current order we're operating on.
- `pending` -> the aggregate value of all of the pending orders.

`currentPosition` is the odd one out here that's not really something that should be loaded in the Context, but is for now for cleanliness since it's used throughout. Since we need this we differentiate between `position` and `currentPosition` with a `latest` prefix.

**Review by kbrizzle — APPROVED** [review]

**Pull request merged**